### ISSUE TITLE
this can't work on macOS Big Sur

### ISSUE THREAD

**Issue by RIRUa** [cocos2d/cocos2d-x#20608]
- cocos2d-x version:4.0
- devices test on:MacBook Pro 
- developing environments
   - NDK version:--
   - Xcode version:12.2
   - VS version:--
   - browser type and version:--

Steps to Reproduce:

1. cocos new test -l cpp --portrait
2. use cmake to test
3. run
4. 2errors is appeared. (about: /usr/lib/libz.dylib & /usr/lib/libiconv.dylib)

**Comment by codetypess** [comment]
cmake/Modules/CocosConfigDepend.cmake

use find_library:
   find_library(ICONV_LIBRARY iconv)
   find_library(Z_LIBRARY z)

**Comment by neonguyen** [comment]
@zhongfq I did your tutorial and get another error
This one is on a real device:
![Screen Shot 2020-11-24 at 22 20 17](https://user-images.githubusercontent.com/8515901/100114809-617c9180-2ea4-11eb-8b4b-95e6307ff9b2.png)
and this one is on a simulator:
![Screen Shot 2020-11-24 at 21 51 46](https://user-images.githubusercontent.com/8515901/100114968-8d981280-2ea4-11eb-89ae-403ff076695e.png)

Note: I did change the "Build Active Architecture Only" to "No"

Not only that I see a lot of deprecated functions. Should we think to remove all of them??? The almost iOS device now is running iOS > 12.  Should we think to stop support iOS < 12??????

**Comment by codetypess** [comment]
@neonguyen you should set some Xcode property in project CMakeLists.txt:
    set_xcode_property(${APP_NAME} SKIP_INSTALL "NO")
    set_xcode_property(${APP_NAME} INSTALL_PATH "$(LOCAL_APPS_DIR)")

**Comment by codetypess** [comment]
@neonguyen in CocosConfigDepend.cmake for iOS:
# Locate system libraries on iOS
find_library(UIKIT_LIBRARY UIKit)
find_library(OPENGLES_LIBRARY OpenGLES)
find_library(CORE_MOTION_LIBRARY CoreMotion)
 find_library(AVKIT_LIBRARY AVKit)
 find_library(CORE_MEDIA_LIBRARY CoreMedia)
find_library(CORE_TEXT_LIBRARY CoreText)
find_library(SECURITY_LIBRARY Security)
find_library(CORE_GRAPHICS_LIBRARY CoreGraphics)
find_library(AV_FOUNDATION_LIBRARY AVFoundation)
find_library(WEBKIT_LIBRARY WebKit)
find_library(ICONV_LIBRARY iconv)
find_library(Z_LIBRARY z)
list(APPEND PLATFORM_SPECIFIC_LIBS
                 ${UIKIT_LIBRARY}
                 ${OPENGLES_LIBRARY}
                 ${CORE_MOTION_LIBRARY}
                 ${AVKIT_LIBRARY}
                 ${CORE_MEDIA_LIBRARY}
                 ${CORE_TEXT_LIBRARY}
                 ${SECURITY_LIBRARY}
                 ${CORE_GRAPHICS_LIBRARY}
                 ${AV_FOUNDATION_LIBRARY}
                 ${WEBKIT_LIBRARY}
                 ${COCOS_APPLE_LIBS}
                 ${ICONV_LIBRARY}
                 ${Z_LIBRARY}
                 )

**Comment by neonguyen** [comment]
@zhongfq I did set all those things. but It still did not work. 
My environments:
Macbook pro 15" 2019
Cocos2dx: 4.0 (change from 3.x by pull request)
XCode version 12.2
MacOS: big sur
CMake: 3.18.6

**Comment by codetypess** [comment]
@neonguyen my env as same as you too, you can try my fork: https://github.com/zhongfq/cocos-lua, and then you can compare what different, especially the cocos folder

**Comment by neonguyen** [comment]
@zhongfq one more thing, I am using python 3.7..6. Can it affect the cocos project? and I am using C++

**Comment by codetypess** [comment]
@neonguyen no. my project base on c++, i use command 'cocos new -l cpp' create my project. i think there are miss some Xcode property in your project. Your program i had meet before

**Comment by neonguyen** [comment]
> @neonguyen no. my project base on c++, i use command 'cocos new -l cpp' create my project. i think there are miss some Xcode property in your project. Your program i had meet before

Thanks for your time. 
let me check your fork and feedback later.

**Comment by neonguyen** [comment]
@zhongfq I tried again a few times. Unfortunately, It still did not work. My V4 version is upgraded from V3. I worry it can be the reason. I cannot reinstall my Mac now.

**Comment by neonguyen** [comment]
@zhongfq I found the solution to solve the build problem. Now I can run the project on my iPhone. Unfortunately, I cannot run it on the simulator. Can you run it on the simulator?
